src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -329,6 +330,9 @@ EnsureCitusTableCanBeCreated(Oid relationOid)
 	EnsureCoordinator();
 	EnsureRelationExists(relationOid);
 	EnsureTableOwner(relationOid);
+	Relation relation = relation_open(relationOid, AccessShareLock);
```

Comment:
To be more inline with the above, maybe move the code `EnsureTableNotTemporary` or such